onnx/defs/logical/defs.cc

```diff
@@ -7,39 +7,43 @@
 namespace ONNX_NAMESPACE {
 
 inline void unaryLogicalOpInference(InferenceContext& ctx) {
-    // Type inference
-    updateOutputElemType(ctx, 0, TensorProto::BOOL);
-    // Shape inference
-    if (hasInputShape(ctx, 0)) {
-        propagateShapeFromInputToOutput(ctx, 0, 0);
-    }
+  // Type inference
+  updateOutputElemType(ctx, 0, TensorProto::BOOL);
+  // Shape inference
+  if (hasInputShape(ctx, 0)) {
+    propagateShapeFromInputToOutput(ctx, 0, 0);
+  }
 }
 
 std::function<void(OpSchema&)> BinaryLogicDocGenerator(const char* name) {
   return [=](OpSchema& schema) {
+#ifndef __ONNX_NO_DOC_STRINGS
     std::string doc = R"DOC(
 Returns the tensor resulted from performing the `{name}` logical operation
 elementwise on the input tensors `A` and `B` (with Numpy-style broadcasting support).
 
 {broadcast_doc}
 )DOC";
-        ReplaceAll(doc, "{name}", name);
```

Comment:
how do you think of adding macro on the declaration/definition of this "replaceall" function? so that in the case of "__ONNX_NO_DOC_STRING" defined, there's no such function for developers to call. if it's still called, there will be compilation error reminding that they should not call it.

Reply:
I was thinking how to avoid people keep adding more such descriptions and we'll keep doing this job of removing them :)

Reply:
maybe, this mechanism should be done on the "SetDoc" API, which is more obvious and stable.

Reply:
Yeah, I'm not super happy with this ReplaceAll setup either. We would rather not visit this issue every time :). I'll think of something. The ReplaceAll setup is also not uniform in the way the replacement is done.

Reply:
Updated.